lib/model/query/forms.js

```diff
@@ -168,11 +168,72 @@ update.audit = (form, data) => (log) => log('form.update', form, { data });
 
 const _updateDef = (form, data) => ({ one }) => one(updater(form.def, data));
 
-const del = (form) => ({ run, Assignments }) =>
-  Promise.all([ run(markDeleted(form)), Assignments.revokeByActeeId(form.acteeId) ]);
+const del = (form) => ({ run, Assignments, Forms }) =>
+  Promise.all([ run(markDeleted(form)), Assignments.revokeByActeeId(form.acteeId) ])
+    .then(() => ((form.currentDefId === null)
+      ? Forms.purge(true, form.id)
+      : null));
 del.audit = (form) => (log) => log('form.delete', form);
 
 
+////////////////////////////////////////////////////////////////////////////////
+// PURGING SOFT-DELETED FORMS
+
+// The main ways we'd want to choose forms to purge are
+// 1. by their deletedAt date (30+ days in the past)
+//    (this would be the default behavior of a purge cron job)
+// 2. if deletedAt is set at all
+//    (useful to purge forms immediately for testing or other time sensitive scenarios)
+// 3. by a specific form id if deletedAt is also set (again for testing or potential future scenarios)
+
+const DAY_RANGE = 30;
+const _trashedFilter = (force, id) => {
+  const idFilter = (id
+    ? sql`and forms.id = ${id}`
+    : sql``);
+  return (force
+    ? sql`forms."deletedAt" is not null ${idFilter}`
+    : sql`forms."deletedAt" < current_date - cast(${DAY_RANGE} as int) ${idFilter}`);
+};
+
+// Purging steps
+// 1. Redact notes about forms from the audit table that reference a form
+//    (includes one kind of comment on a submission)
+// 2. Log the purge in the audit log with actor not set because purging isn't accessible through the api
+// 3. Update actees table for the specific form to leave some useful information behind
+// 4. Delete the forms and their resources from the database
+// 5. Purge unattached blobs
+const purge = (force = false, id = null) => ({ run, all, Blobs }) =>
+  run(sql`
+update audits set notes = ''
+from forms
+where audits."acteeId" = forms."acteeId"
+and ${_trashedFilter(force, id)}`)
+    .then(() => run(sql`
+insert into audits ("action", "acteeId", "loggedAt")
+select 'form.purge', "acteeId",  clock_timestamp()
+from forms
+where ${_trashedFilter(force, id)}`))
+    .then(() => run(sql`
+update actees set "purgedAt" = clock_timestamp(),
+  "details" = json_build_object('projectId', forms."projectId",
+                                'formId', forms.id,
+                                'xmlFormId', forms."xmlFormId",
+                                'name', form_defs."name",
```

Comment:
Have a name field (text vs. varchar) on the actees table, purgedName, nameOnPurged
capture the name explicitly on deleted actees